spec/ios_l10n_helper_spec.rb

```diff
@@ -160,4 +160,82 @@ def file_encoding(path)
+        # Arrange
+        stub = stub_request(:get, "#{gp_fake_url}/fr/default/export-translations?format=strings").to_return(body: 'content')
+        dest = StringIO.new
+        # Act
+        described_class.download_glotpress_export_file(project_url: gp_fake_url, locale: 'fr', filters: nil, destination: dest)
+        # Assert
+        expect(stub).to have_been_made.once
+        expect(dest.string).to eq('content')
+      end
+
+      it 'passes the expected params when a list of filters is provided' do
+        # Arrange
+        stub = stub_request(:get, "#{gp_fake_url}/fr/default/export-translations?format=strings&filters%5Bstatus%5D=current&filters%5Bterm%5D=foobar").to_return(body: 'content')
+        dest = StringIO.new
+        # Act
+        described_class.download_glotpress_export_file(project_url: gp_fake_url, locale: 'fr', filters: { status: 'current', term: 'foobar' }, destination: dest)
+        # Assert
+        expect(stub).to have_been_made.once
+        expect(dest.string).to eq('content')
+      end
+    end
+
+    describe 'destination parameter' do
+      # it 'accepts an IO (File)' # Not really needed, as all previous tests already cover this by using a `StringIO`
+
+      it 'accepts a String (path)' do
+        Dir.mktmpdir('a8c-release-toolkit-tests-') do |tmp_dir|
+          # Arrange
+          # Note: in practice it seems that GlotPress's `.strings` exports are using UTF8 (but served as `application/octet-stream`)
+          #       but it does not hurt to ensure the download to a file can work with UTF16 ()and copy the binary stream verbatim)
+          body = File.read(fixture('Localizable-utf16.strings'))
+          stub = stub_request(:get, "#{gp_fake_url}/fr/default/export-translations?format=strings").to_return(body: body)
+          dest = File.join(tmp_dir, 'export.strings')
+          # Act
+          described_class.download_glotpress_export_file(project_url: gp_fake_url, locale: 'fr', filters: nil, destination: dest)
+          # Assert
+          expect(stub).to have_been_made.once
+          expect(File).to exist(dest)
+          expect(File.read(dest)).to eq(body)
+        end
+      end
+    end
+
+    describe 'invalid parameters' do
+      it 'prints an `UI.error` if passed a non-existing locale (or any other 404)' do
+        # Arrange
+        stub = stub_request(:get, "#{gp_fake_url}/invalid/default/export-translations?format=strings").to_return(status: [404, 'Not Found'])
+        error_messages = []
+        allow(FastlaneCore::UI).to receive(:error) { |message| error_messages.append(message) }
+        dest = StringIO.new
+        # Act
+        described_class.download_glotpress_export_file(project_url: gp_fake_url, locale: 'invalid', filters: nil, destination: dest)
+        # Assert
+        expect(stub).to have_been_made.once
+        expect(error_messages).to eq(['Error downloading locale `invalid` — 404 Not Found'])
```

Comment:
Yep, exactly: this expectation not only check that I have the one `UI.error`, but also ensures that we didn't get any other, unexpected ones, but only the one I expect.

Reply:
This actually just saved my bacon as I fixed something else in the implementation and made a typo, as this expectation then caught a second error message that I didn't expect to be there (and was because of said typo). So that helped me fix the implementation change I made  in order to make that test go green again and return to only one error message  💪